source/Symbol/SwiftASTContext.cpp

```diff
@@ -2950,6 +2950,21 @@ swift::ASTContext *SwiftASTContext::GetASTContext() {
       m_ast_context_ap->addModuleLoader(std::move(serialized_module_loader_ap));
     }
 
+    // Install the parseable interface module loader.
```

Comment:
I think this one has to go before the serialized one in order to get the correct fallback behavior when the binary swiftmodule is invalid / unreadable / out of date. I guess there should be a test for that too!

Reply:
Good catch – thank you! Moved it up and added a version of the tests that adds empty .swiftmodule files corresponding to each of the libs beforehand.